bcdaworker/main.go

```diff
@@ -162,6 +158,33 @@ func appendErrorToFile(acoID, code, detailsCode, detailsDisplay string) {
 	}
 }
 
+func jsonToNDJSON(w *bufio.Writer, jsonData, jsonType, beneficiaryID, acoID string) {
+	var jsonOBJ map[string]interface{}
+	err := json.Unmarshal([]byte(jsonData), &jsonOBJ)
+	if err != nil {
+		log.Error(err)
+		appendErrorToFile(acoID, responseutils.Exception, responseutils.InternalErr, fmt.Sprintf("Error UnMarshaling %s from data for beneficiary %s in ACO %s", jsonType, beneficiaryID, acoID))
+		return
+	}
+
+	entries := jsonOBJ["entry"]
+	for _, entry := range entries.([]interface{}) {
+		entryJSON, err := json.Marshal(entry)
```

Comment:
This is probably me just not knowing Go (at all!) but it doesn't look like you're actually doing anything with the data you've marshalled here?

Reply:
entries is the in memory map of just the values within entry.  I have to marshal that back to JSON before I write it out to the file(line 179).  I can't just do string(entry) because it wouldn't be in the right format.

Reply:
Is there a streaming JSON processor available for Go that could do this without marshalling the whole thing to RAM? Some of these are going to be... big.

If not, I think this section of the code needs a comment explaining that this harmless-looking line could eat tons of RAM (especially when run concurrently across bunches of goroutines!):

    json.Unmarshal([]byte(jsonData), &jsonOBJ)

(Suggestion: please follow up and ask something like, "Backend peeps, do we know how big we expect those NDJSON files to be in the average, 90% and worst cases?" Preliminary answer that's framed in terms of number of EOBs rather than bytes: "Claim Counts" on <https://confluence.cms.gov/display/BB/Blue+Button+Numbers>)